docs/design/flow-traffic-mirroring/flow-traffic-mirroring.md

```diff
@@ -0,0 +1,171 @@
+      "mirror_point_switch_id": string,
+      "sink_endpoint": {
+         "switch_id": string,
+         "port": int,
+         "inner_vlan": int,
+         "outer_vlan": int
+      }
+  }
+  ```
+
+* Delete mirror point:
+
+  `DELETE /flows/{flow_id}/mirror/{mirror_point_id}`
+
+
+* Get list of mirror points by flow id:
+
+  `GET /flows/{flow_id}/mirror`
+
+  Response payload:
+  ```
+  {
+      "flow_id": string,
+      "points":[
+          {
+              "mirror_point_id": string,
+              "mirror_direction": string [FORWARD|REVERSE],
+              "mirror_point_switch_id": string,
+              "sink_endpoint": {
+                  "switch_id": string,
+                  "port": int,
+                  "vlan": int
+              }
+          }
+      ]
+  }
+  ```
+
+
+* API that needs to be updated: 
+  - Need to add information about the state of the mirror paths in the flow payload.
+  - It is necessary to add information about built mirror paths to API `GET /flows/{flow_id}/path`
+
+## Workflow
+
+Creation and deletion flow mirror points in the FlowHSTopology:
+
+![Create flow traffic mirror point](./create-mirror-point.png "Create flow traffic mirror point")
+![Delete flow traffic mirror point](./delete-mirror-point.png "Delete flow traffic mirror point")
+
+Getting flow mirror points in the NbWorkerTopology:
+
+![Get flow traffic mirror point](./get-mirror-point.png "Get flow traffic mirror point")
+
+## Affected kilda components
+* need to add mirror groups to the Floodlight;
+* need to update RerouteTopology to react to the network events for the mirror paths;
+* add logic to the flow update operation when updating flow endpoints;
+* update switch and flow validation.
+
+## Limitations
+It is allowed to use a transit switch as a mirror point only if flow is pinned.
+
+## Related issues
+There are currently no asymmetric bandwidth ISLs in the system. This feature sets paths in one direction only. 
+This results in asymmetric ISLs appearing in the system. This effect on the system requires more in-depth research.
+
+A possible solution to this issue is to create a dummy path that will consume bandwidth in the opposite direction.
+
+## Switch rules
+Existing actions set for any type of existing OF flows (ingress, transit, egress) 
+will be replaced with a "goto group" action instead of an "output port" action.
+The group will have 2 or more buckets: one will represent output to the flow or ISL port, 
+the rest will represent mirror actions set (i.e. routing to the mirror paths).
+
+Examples of rules for VLAN transit encapsulation:
+* _Group_:
+```
+{
+   "dpid": dpid,
+   "type": "ALL",
+   "group_id": GROUP_ID,
+   "buckets":[
+      {
+         "actions":[
+            {
+               "type": "OUTPUT",
+               "port": flow_or_isl_out_port
+            }
+         ]
+      },
+      {
+         "actions":[
+            {
+               "type": "SET_FIELD", 
+               "field": "vlan_vid",
```

Comment:
Will/should it work with vxlan transit encoding?

Reply:
These examples have been removed as they are confusing. An explanation that we will use transit vlan encapsulation in the same way as for flow was added to the document.